C/hal/PB/PB.c

```diff
@@ -0,0 +1,39 @@
+#include "../../common/Types.h"
+#include "../../mcal/GPIO/GPIO.h"
+#include "PB.h"
+
+void PB_Init(PB_t* pPB)
+{
+    GPIO_SetPinDirection(pPB->Port, pPB->Pin, pPB->State);
+}
+
+PressedState_t PB_GetPressed(PB_t* pPB)
+{
+    state_t LocPinValue = 0;
```

Comment:
don't use `loc`, .. as prefix just `pinValue`
All of us really know thats a local variable